_P248_TempHumidity_AHT1x.ino

```diff
@@ -0,0 +1,339 @@
+      Device[deviceCount].PullUpOption = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption = true;
+      Device[deviceCount].ValueCount = 2;
+      Device[deviceCount].SendDataOption = true;
+      Device[deviceCount].TimerOption = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_248);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_248));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_248));
+      //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_248));
+      //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[3], PSTR(PLUGIN_VALUENAME4_248));
+    //  strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[4], PSTR(PLUGIN_VALUENAME5_248));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+    {
+      byte choice = PCONFIG(0);
+
+      int optionValues[2] = { 0x38, 0x39 };
+      addFormSelectorI2C(F("i2c_addr"), 2, optionValues, choice);
+      addFormNote(F("AO Low=0x38, High=0x39"));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      PCONFIG(0) = getFormItemInt(F("i2c_addr"));
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_INIT:
+    {
+      String log;
+      uint8_t addr = PCONFIG(0);
+
+      if (!AHT10_init(addr))
+        log = F("AHT10: init");
+
+      //if (!AHT10_setNormalMode(addr))         //one measurement+sleep mode
+      //  log = F("AHT10: setNormalMode");
+      //if (!AHT10_enableFactoryCalCoeff(addr)) //load factory calibration coeff
+       // log = F("AHT10: enableFactoryCoeff");
+
+      /*check calibration enable */
+      //if (AHT10_getCalibrationBit(false) != 0x01) 
+       // log = F("AHT10: getCalibrationBit");
+
+      if (log)
+        addLog(LOG_LEVEL_ERROR,log);
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      uint8_t addr = PCONFIG(0);
+      AHT1XStartMeasurement(addr);
+      delay(AHT10_MEASURMENT_DELAY);
+      bool ret=AHT1XReadMesurement(addr);
+      if (!ret)
+      {
+        String err = F("AHT1x: read error: ");
+        err+= ret;
+        addLog(LOG_LEVEL_ERROR,err);
+      }
+
+      UserVar[event->BaseVarIndex + 0] = AHT10_getTemperature();
+      UserVar[event->BaseVarIndex + 1] = AHT10_getHumidity();
+      //UserVar[event->BaseVarIndex + 2] = Plugin_248_SHT2x->abshum;
+      //UserVar[event->BaseVarIndex + 3] = Plugin_248_SHT2x->doVent;
+      
+      String log = F("AHT1x: Temperature: ");
+      log += UserVar[event->BaseVarIndex + 0];
+      addLog(LOG_LEVEL_INFO, log);
+      log = F("AHT1x: Humidity: ");
+      log += UserVar[event->BaseVarIndex + 1];
+      addLog(LOG_LEVEL_INFO, log);
```

Comment:
Check for `logLevelActiveFor` wrapping this log stuff.

Reply:
ok

Reply:
I will add. However isn't better to make that check part of the log function ?

Reply:
When adding a check it will avoid ('expensively') building the log string that is never used, so it would have to capture the entire relevant code.